samcli/lib/providers/api_provider.py

```diff
@@ -51,44 +48,43 @@ def get_all(self):
 
         yield self.api
 
-    def _extract_api(self, resources):
+    def _extract_api(self):
         """
         Extracts all the routes by running through the one providers. The provider that has the first type matched
         will be run across all the resources
 
         Parameters
         ----------
-        resources: dict
-            The dictionary containing the different resources within the template
         Returns
         ---------
         An Api from the parsed template
         """
 
         collector = ApiCollector()
-        provider = self.find_api_provider(resources)
-        provider.extract_resources(resources, collector, cwd=self.cwd)
+        provider = self.find_api_provider(self.stacks)
```

Comment:
better to call the static method through the class instead of the instance, i.e., `ApiProvider.find_api_provider(self.stacks)`

Reply:
I am not sure which it is better? I don't find doc online regarding which one is better. In official doc, it writes:

> A static method can be called either on the class (such as C.f()) or on an instance (such as C().f()).

In my own opinion, calling static methods in the same class with `self.` provides a hint that this method is in the same file, and it is shorter to write. But I think it does not matter too much, what do you think?

Reply:
when I read this line at the first glance, I thought `find_api_provider` is an instance method and had the question: why do we send `self.stacks` as a parameter while we already have access to `self`. If it was called with the class name instead, I wouldn't need to go down to the method to realize that it is not an instance method.

Reply:
it makes sense, I will change them in the final PR, there are other places with this usage as well. Thx

Reply:
done in https://github.com/aws/aws-sam-cli/pull/2662/commits/31d4d4ad1852d1020cfa67f9f52d5f29ef428136